### ISSUE TITLE
Pattern Match for Round Parentheses using `body_replace_all_text()`

### ISSUE THREAD

**Issue by DragonflyStats** [davidgohel/officer#439]
Hi David
I am trying to replace some text that includes a round parenthesis bracket. The intended change is similar to the `gsub()` operation. I cant figure out the regular expression match for the text in the bracket (OLDTEXT in the code snippet below) when working with `body_replace_all_text()`. Is this something specific to the {officer} package?

<pre><code>
library(officer)
library(magrittr)

OLDTEXT <- "Limerick (Ireland)"
NEWTEXT <- "Limerick, Ireland"

gsub(TEST1,NEWTEXT,"The location is Limerick (Ireland)")

doc <- read_docx()
doc <- body_add_par(doc, "The location is Limerick (Ireland)")

my_doc <- my_doc %>% body_replace_all_text(OLDTEXT,NEWTEXT)

OLDTEXT2 <- "Limerick \\(Ireland)"

my_doc <- my_doc %>% body_replace_all_text(OLDTEXT2 ,NEWTEXT)

</code></pre>

**Comment by davidgohel** [comment]
Hello,

No I don't think there is anything special here. I think you mixed things between doc and my_doc (I had errors when I tried your code). The following works as expected:

```r
library(officer)
library(magrittr)

OLDTEXT <- "Limerick \\(Ireland\\)"
NEWTEXT <- "Limerick, Ireland"

doc <- read_docx() %>% 
  body_add_par("The location is Limerick (Ireland)") %>% 
  body_replace_all_text(OLDTEXT ,NEWTEXT)

print(doc, target = "coco.docx")

```

<img width="632" alt="Capture d’écran 2022-07-20 à 13 58 16" src="https://user-images.githubusercontent.com/4331618/179976350-c1acdc63-6d42-43b2-b420-3986cb8d269c.png">

**Comment by DragonflyStats** [comment]
Hi David 
Thanks for that. The confusion between "doc" and "my_doc"isn't the issue. I suspect the real issue is something else entirely, and is just manifesting here. I am processing multiple documents. <tt>body_replace_all_text()</tt> is catching the OLDTEXT in some documents, but occasionally missing the same OLDTEXT in others - giving the "no matches found". 
I will re-install everything, and see how I get on. You can close the issue. If you have time sometime - I can show you the actual workflow over a zoom call - but we can wait till a cold winter evening to do that.
Thanks 
Kevin

**Comment by davidgohel** [comment]
> occasionally missing the same OLDTEXT in others - giving the "no matches found"

OK, I did not understand that – I thought you were asking about pattern and regular expression. It's important if you can provide a reproducible example.   

I think you are facing the issue with `chunking of text`, see section "Chunking of text" here: https://davidgohel.github.io/officer/reference/body_replace_all_text.html

**Comment by DragonflyStats** [comment]
Hi David
I didn't understand it at the time either. I thought the problem was my under-developed RegEx skills - but there are some peculiar behaviours elsewhere that now makes me think the problem is something else. It is hard to reproduce the issue here. I think a demonstration would be necessary for proper inspection.

I have a look at that reference that you suggest.

 We have to replace some text in about 500 word documents. Overall the job went well - and we got through it very quickly.

**Comment by davidgohel** [comment]
For example, in Word, if you type "Cowa" then "bunga", it is likely that Word will store two chunks, one for "Cowa" and one for "bunga", even if displayed as one single word. If that happen, `body_replace_all_text()` will not be able to find and replace "Cowabunga". 

I am working on #433 (almost ready!), would properties fields usage be a possible solution for you? Are these 500 docs manually edited, or are they automatically generated? If sometimes it fails, I guess some manual typing have been done?

**Comment by DragonflyStats** [comment]
Yes I have read that documentation and it seems I was wrong about the cause of the problem. Your example does indeed fit the description of the problem. 

The documents were prepared with {officer} - but then editted manually by scientists in the interim period. The vast majority were corrected with new text - and I was able to detect the few did not take the corrections, and can fix them manually.
This is my first case of working with pre-existing documents with {officer}, whereas before I would be the original creator,  so I just have to improve my skills accordingly.
